### PR TITLE
Make STS available replicas optional

### PR DESCRIPTION
**Pull request by ashrayjain**
#### Which issue(s) this PR fixes:
Fixes https://github.com/kubernetes/kubernetes/issues/109210

#### Does this PR introduce a user-facing change?
```release-note
Fixes a 1.22 regression in published openapi: make STS available replicas optional again
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
Make STS available replicas optional again.
```

### CODE REVIEW

**Comment by ashrayjain** [comment]
@liggitt this PR is ready for your review! Thanks

File: staging/src/k8s.io/api/apps/v1/types.go [hunk 1]
```diff
@@ -272,6 +272,7 @@ type StatefulSetStatus struct {
 
 	// Total number of available pods (ready for at least minReadySeconds) targeted by this statefulset.
 	// This is a beta field and enabled/disabled by StatefulSetMinReadySeconds feature gate.
+	// +optional
```
Comment (FillZpp) [inline staging/src/k8s.io/api/apps/v1/types.go:275]: And `omitempty` should be added into the json tag. So do the beta apis.

Comment (liggitt) [reply staging/src/k8s.io/api/apps/v1/types.go:275]: no, we only want to change the openapi schema requiredness, not any serialization

**Comment by xmudrii** [comment]
@ashrayjain @liggitt Just to confirm, this has been fixed in 1.24 by #109241, so we only need to cherry-pick it to 1.23. Is that correct?

**Comment by liggitt** [comment]
yes

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**